internal/devconfig/config.go

```diff
@@ -63,8 +78,106 @@ func IsDefault(path string) bool {
 	return cfg.Root.Equals(&DefaultConfig().Root)
 }
 
-func LoadForTest(path string) (*Config, error) {
-	return readFromFile(path)
+// Open loads a Devbox config from a file or project directory. If path is a
+// directory, Open looks for a well-known config name (such as devbox.json)
+// within it. The error will be [ErrNotFound] if path is a valid directory
+// without a config file.
+//
+// Open does not recursively search outside of path. See [Find] to load a config
+// by walking up the directory tree.
+func Open(path string) (*Config, error) {
+	start := time.Now()
```

Comment:
If you want to profile performance, you can use `defer debug.FunctionTimer().End()`

Reply:
Maybe we can log that to slog as well?